cirque/capabilities/networksimulationcapability.py

```diff
@@ -0,0 +1,48 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import os
+import glob
+import subprocess
+from threading import Lock
+
+from cirque.capabilities.basecapability import BaseCapability
+
+
+class NetworkSimulationCapability(BaseCapability):
+
+    def __init__(self, latencyMs):
+        self.latencyMs = latencyMs
+
+    @property
+    def name(self):
+        return 'NetworkSimulation'
+
+    @property
+    def description(self):
+        return {
+            'latency': self.latencyMs,
+        }
+
+    def get_docker_run_args(self, dockernode):
+        return {
+            'cap_add': ['NET_ADMIN']
+        }
+
+    def enable_capability(self, docker_node):
+        docker_node.container.exec_run("tc qdisc add dev eth0 root netem delay {}ms".format(self.latencyMs))
```

Comment:
This capability can add delay for current docker node for any imbound/outbound packets?